app/models/workflow_step.rb

```diff
@@ -11,25 +11,51 @@ class WorkflowStep < ApplicationRecord
   validate  :workflow_exists, on: :create
   validate  :process_exists_for_workflow, on: :create
 
+  before_save :set_completed_at, if: :completed?
+
   scope :lifecycle, -> { where.not(lifecycle: nil) }
-  scope :incomplete, -> { where.not(status: %w[completed skipped]) }
-  scope :complete, -> { where(status: %w[completed skipped]) }
+  scope :incomplete, -> { where.not(status: COMPLETED_STATES) }
+  scope :complete, -> { where(status: COMPLETED_STATES) }
```

Comment:
Some refactoring here to reduce duplication, and because I added a new helper method to ask if the step is completed.